src/ui/graph.rs

```diff
@@ -545,12 +545,12 @@ impl Graph {
             }
 
             let mut pointer_delta = egui::Vec2::ZERO;
-            if ui.ui_contains_pointer() // Can only be queried after the editor UI has been drawn
+            let is_secondary_button_down = ui.ui_contains_pointer() // Can only be queried after the editor UI has been drawn
                 && ui.input(|i| {
                     pointer_delta = i.pointer.delta();
                     i.pointer.secondary_down()
-                })
-            {
+                });
+            if is_secondary_button_down {
                 self.editor.pan_zoom.pan += pointer_delta;
             }
```

Comment:
I'd like to keep the laziness from short-circuiting and also avoid mutating `pointer_delta` from the closure that's supposed to just check for secondary down
```suggestion
            // Can only be queried after the editor UI has been drawn
            if ui.ui_contains_pointer() {
                let (secondary_down, pointer_delta) =
                    ui.input(|i| (i.pointer.secondary_down(), i.pointer.delta()));

                if secondary_down {
                    self.editor.pan_zoom.pan += pointer_delta;
                }
            }
```

Reply:
It does not change the lazyness of the operation: if `ui.ui_contains_pointer()` returns false, then the `ui.input` is not called. Assigning the result to a variable does not change the short-circuiting rules.

Reply:
You're probably right, it's just that there are locks and atomics involved behind the scenes, and also it's clearer in the code that we only need the data if the ui contains the pointer. Regardless, it's better to get `pointer_delta` the same way as `secondary_down` instead of mutating it

Reply:
Fair enough, you are right that it is much cleaner in this way :blush:

I am a bit in a hurry in this moment, I will make the changes later. Thank you for your time :heart: